dev/tasks/conda-recipes/arrow-cpp/build-pyarrow.sh

```diff
@@ -39,11 +39,6 @@ else
     export PYARROW_WITH_CUDA=0
 fi
 
-# Resolve: Make Error at cmake_modules/SetupCxxFlags.cmake:338 (message): Unsupported arch flag: -march=.
```

Comment:
@kou which tests shall we run against this change?

Reply:
`conda-linux-*` and `conda-osx-*` but we should wait for #14102.